qiskit_experiments/calibration/update_library.py

```diff
@@ -0,0 +1,176 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A library of experiment calibrations."""
+
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import List, Tuple, Union
+import numpy as np
+
+from qiskit.circuit import Parameter
+from qiskit.pulse import ScheduleBlock
+
+from qiskit_experiments.experiment_data import ExperimentData
+from qiskit_experiments.calibration.backend_calibrations import BackendCalibrations
+from qiskit_experiments.calibration.calibrations import Calibrations
+from qiskit_experiments.calibration.parameter_value import ParameterValue
+from qiskit_experiments.calibration.exceptions import CalibrationError
+from qiskit_experiments.calibration.calibration_key_types import ParameterValueType
+
+
+class BaseUpdater(ABC):
+    """A base class to update calibrations."""
+
+    @staticmethod
+    def _time_stamp(exp_data: ExperimentData) -> datetime:
+        """Helper method to extract the datetime."""
+        all_times = exp_data.completion_times.values()
+        if all_times:
+            return max(all_times)
+
+        return datetime.now()
+
+    @classmethod
+    def _add_parameter_value(
+        cls,
+        exp_data: ExperimentData,
+        cal: Calibrations,
+        value: ParameterValueType,
+        param: Union[Parameter, str],
+        schedule: Union[ScheduleBlock, str] = None,
+        group: str = "default",
+    ):
+        """Update the calibrations with the given value.
+
+        Args:
+            exp_data: The ExperimentData instance that contains the result and the experiment data.
+            cal: The Calibrations instance to update.
+            value: The value extracted by the subclasses in the :meth:`update` method.
+            param: The name of the parameter, or the parameter instance, which will receive an
+                updated value.
+            schedule: The ScheduleBlock instance or the name of the instance to which the parameter
+                is attached.
+            group: The calibrations group to update.
+        """
+
+        qubits = exp_data.data(0)["metadata"]["qubits"]
+
+        param_value = ParameterValue(
+            value=value,
+            date_time=BaseUpdater._time_stamp(exp_data),
```

Comment:
Sorry I found one more thing to update
```suggestion
            date_time=cls._time_stamp(exp_data),
```

Note that if subclass overrides the class method, this is not reflected because this is always called via the `BaseUpdater`. Usually it's better to use `cls` to access to the class method.